internal/condition/eval/eval.go

```diff
@@ -25,56 +26,69 @@ func EvaluateTupleCondition(
 	tupleKey *openfgav1.TupleKey,
 	typesys *typesystem.TypeSystem,
 	context *structpb.Struct,
-) (*condition.EvaluationResult, error) {
```

Comment:
(nit non-blocking): I'm not in favor of the changes around the return type, but I don't want to block these changes.

For context - I think it's misleading to return a bool as that can blindly imply to the caller that the condition was evaluable and resolved to `false`. A nil EvaluationResult implies an error in trying to resolve an evaluation outcome, which is distinctly different than returning a concrete bool with an overloaded error. 

The intent behind having the explicit EvaluationResult was 2-fold: 1) it allows us to grow the response of the evaluation outcome, which would include a residual expression in the future (e.g. we can evaluate a sub-branch of the CEL expression but we return a residual unresolvable expression). And 2) it's the callers responsibility to determine what to do with the outcome of the CEL expression evaluation. It's the `eval` package's responsibility to have a clean API boundary for callers to use the information about the evaluation in a way that makes sense to the caller. The way the code is written in these changes implies something about the callers behavior, and so that feels like a leak of responsibility. 

In general the outcome of a CEL evaluation is a tertiary outcome, not a binary one. A CEL expression can evaluate to a binary value if it is resolvable (e.g. `x && y`, where `x=false` or `x || y`, where `y=true`) or it can result in a residual expression (AST), for example `(x || y) && z`, where x=true and y and z are not provided - residual is the expression `true && z` .

Given we're not really leveraging this distinction in this tertiary state I don't want to block this, but I did want to mention that I think this is the wrong direction for this internal libraries return signature.